interview-questions.md

```diff
@@ -9,35 +9,65 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 3. Name three rails generator commands. What is created by each?
 
 Your answer:
+I think we have to do something to make the CRUD actions. We have r generate, then I think theres the resource one but I can't remember what it was called. 
 
 Researched answer:
+There are three useful Rails generator commands. The first one is "rails generate model," which creates a new model file along with its corresponding migration file. This command helps me generate the skeletal structure of a model, including its attributes and associations. The second command is "rails generate controller," which generates a new controller file that handles the logic and actions related to a specific resource. It creates the basic CRUD actions, along with corresponding views and test files. Lastly, the "rails generate scaffold" command is quite handy as it creates a complete set of resources, including the model, migration, controller, views, and routes. It saves me time by automatically generating the necessary files for a basic CRUD interface. These generator commands simplify the process of setting up different components in a Rails application and enhance my productivity as a developer. Though we have been warned to not use this command in class because it can cause some major problems.
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
 action: "GET" location: /students
-
+This method would retrieve and display a list of all students.
 action: "POST" location: /students
-
+This method would handle the creation of a new student based on the provided parameters in the request.
 action: "GET" location: /students/new
-
+This method would render a form to create a new student.
 action: "GET" location: /students/2
-
+This method would retrieve and display the details of the student with the ID of 2.
 action: "GET" location: /students/2/edit
-
+This method would render a form to edit the details of the student with the ID of 2.
 action: "PATCH" location: /students/2
-
+This method would handle updating the details of the student with the ID of 2 based on the provided parameters in the request.
 action: "DELETE" location: /students/2
+This method would handle the deletion of the student with the ID of 2.
```

Comment:
This question is asking for the controller methods associated with each action and location given. For example, line 38 has a GET request and the path of /students, so this would be for the `index` controller method.